src/folder/feature-folder-tool-bar/FolderToolBar.tsx

```diff
@@ -0,0 +1,131 @@
+import styles from "./FolderToolBar.module.scss";
+import classNames from "classnames/bind";
+import { AddFolderButton } from "../../folder/ui-add-folder-button";
+import { FolderButton } from "../../folder/ui-folder-button";
+import { IconAndTextButton } from "../../sharing/ui-icon-and-text-button";
+import { ALL_LINKS_TEXT, BUTTONS, KAKAO_SHARE_DATA, MODALS_ID } from "./constant";
+import { ALL_LINKS_ID } from "../../link/data-access-link/constant";
+import { KeyboardEvent, useState } from "react";
+import { ShareModal } from "../../folder/ui-share-modal";
+import { InputModal } from "../../sharing/ui-input-modal";
+import { AlertModal } from "../../sharing/ui-alert-modal";
+import { copyToClipboard } from "../../sharing/util/copyToClipboard";
+import { useKakaoSdk } from "../../sharing/util/useKakaoSdk";
+
+const cx = classNames.bind(styles);
+
+interface Folder {
+  id: number;
+  createdAt: string;
+  name: string;
+  userId: number;
+  linkCount: number;
+};
+
+type SelectedFolderId = number | "all";
+
+interface FolderToolBarProps  {
+  folders: Folder[];
+  selectedFolderId: SelectedFolderId;
+  onFolderClick: (folderId: SelectedFolderId) => void;
+};
+
+export const FolderToolBar = ({ folders, selectedFolderId, onFolderClick }: FolderToolBarProps) => {
+  const { shareKakao } = useKakaoSdk();
+  const [currentModal, setCurrentModal] = useState<string | null>(null);
+  const [inputValue, setInputValue] = useState<string>("");
+
+  const folderName =
+    ALL_LINKS_ID === selectedFolderId
+      ? ALL_LINKS_TEXT
+      : folders?.find(({ id }) => id === selectedFolderId)?.name ?? "";
+  const shareLink = `${window.location.origin}/shared?user=1&folder=${selectedFolderId}`;
+
+  const closeModal = () => setCurrentModal(null);
+  const handleKeyDown = (event: KeyboardEvent<HTMLDivElement>) => {
```

Comment:
❓ 
혹시 typescript가 정확히 동작했는지 궁금합니다. 

```suggestion
  const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
```

Reply:
💊 
`handleKeyDown` 의 실제 사용처를 살펴보니 Modal쪽에서 `esc` 키를 입력 했을 때 꺼지도록 한 거였군요~
`esc` 를 누를때의 관심사는 `ToolBar` 보다는 각 `Modal`에게 책임이 더 있는 것 같습니다

비슷한 모달의 형태가 굉장히 많으니 한번 공용 모달 컴포넌트를 만들어보는건 어떠신지요 ㅎㅎ
어떻게 설계 하면 좋을까~ 에 대한 고민을 하신 뒤 코멘트를 달아드리면 제가 생각하는 방향성도 제안드리도록 하겠습니다.